audio/src/main.zig

```diff
@@ -0,0 +1,120 @@
+const std = @import("std");
+const mem = std.mem;
+const testing = std.testing;
+const builtin = @import("builtin");
+const Backend = switch (builtin.os.tag) {
+    .linux,
+    .windows,
+    .macos,
+    .ios,
+    => @import("soundio.zig"),
+    else => @compileError("unsupported os"),
+};
+pub const Error = Backend.Error;
+pub const Device = Backend.Device;
+pub const DeviceIterator = Backend.DeviceIterator;
+const DeviceDescriptorInternal = Backend.DeviceDescriptorInternal;
+
+pub const Mode = enum {
+    input,
+    output,
+};
+
+pub const Format = enum {
+    U8,
+    S16,
+    S24,
+    S32,
+    F32,
+};
+
+pub const DeviceDescriptor = struct {
+    mode: ?Mode = null,
+    format: ?Format = null,
+    is_raw: ?bool = null,
+    channels: ?u8 = null,
+    sample_rate: ?u32 = null,
+    id: ?[]const u8 = null,
+    name: ?[]const u8 = null,
+    callback: ?fn (device: Device, frame_count: u32) void = null,
+
+    internal: ?DeviceDescriptorInternal = null,
```

Comment:
This is a bad idea. This should be strictly kept inside soundio.zig's Device struct

Reply:
in fact there's no need to move it into somewhere else
i was also doubt to keep it, just waited for reviews